Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/Makefile
===================================================================
--- kernel-rockchip.orig/arch/arm64/boot/dts/rockchip/Makefile
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/Makefile
@@ -132,3 +132,6 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588s-t
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588s-tablet-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588s-tablet-v11.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-firefly-itx-3588j.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += roc-rk3588s-pc.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += roc-rk3588s-pc-mipi101-M101014-BE45-A1.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += roc-rk3588s-pc-dual-mipi101-M101014-BE45-A1.dtb
Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/rk3588-roc-pc-cam-8ms1m.dtsi
===================================================================
--- /dev/null
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/rk3588-roc-pc-cam-8ms1m.dtsi
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+ *
+ */
+
+/ {
+	cam_ircut0: cam_ircut {
+		status = "disabled";
+		compatible = "rockchip,ircut";
+		ircut-open-gpios = <&gpio4 RK_PA6 GPIO_ACTIVE_HIGH>;
+		ircut-close-gpios  = <&gpio4 RK_PA7 GPIO_ACTIVE_HIGH>;
+		rockchip,camera-module-index = <0>;
+		rockchip,camera-module-facing = "back";
+	};
+	vcc_mipidphy0: vcc-mipidcphy0-regulator {
+		status = "disabled";
+		compatible = "regulator-fixed";
+		gpio = <&gpio1 RK_PB1 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&mipidphy0_pwr>;
+		regulator-name = "vcc_mipidphy0";
+		enable-active-high;
+	};
+};
+
+&csi2_dphy0 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipidphy0_in_ucam0: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&xc7160_out0>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			csidphy0_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&mipi2_csi2_input>;
+			};
+		};
+	};
+};
+
+&csi2_dphy0_hw {
+	status = "okay";
+};
+
+&i2c7 {
+        status = "okay";
+        pinctrl-names = "default";
+	pinctrl-0 = <&i2c7m2_xfer>;
+
+        XC7160: XC7160b@1b{
+               compatible = "firefly,xc7160";
+               reg = <0x1b>;
+	       clocks = <&cru CLK_MIPI_CAMARAOUT_M1>;
+               clock-names = "xvclk";
+               pinctrl-names = "default";
+	       pinctrl-0 = <&mipim1_camera1_clk>;
+               power-domains = <&power RK3588_PD_VI>;
+
+               power-gpios = <&gpio4 RK_PB5 GPIO_ACTIVE_LOW>;
+               reset-gpios = <&gpio0 RK_PD5 GPIO_ACTIVE_HIGH>;
+               pwdn-gpios = <&gpio4 RK_PB4 GPIO_ACTIVE_HIGH>;
+
+               //avdd-supply = <&vcc_mipidcphy0>;
+               firefly,clkout-enabled-index = <0>;
+               rockchip,camera-module-index = <0>;
+               rockchip,camera-module-facing = "back";
+               rockchip,camera-module-name = "NC";
+               rockchip,camera-module-lens-name = "NC";
+               port {
+                        xc7160_out0: endpoint {
+                               remote-endpoint = <&mipidphy0_in_ucam0>;
+                               data-lanes = <1 2 3 4>;
+                       };
+               };
+       };
+
+};
+
+&mipi2_csi2 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi2_csi2_input: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&csidphy0_out>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi2_csi2_output: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&cif_mipi2_in0>;
+			};
+		};
+	};
+};
+
+&pinctrl {
+	cam {
+		mipidphy0_pwr: mipidphy0-pwr {
+			rockchip,pins =
+				/* camera power en */
+				<1 RK_PB1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
+&rkcif {
+	status = "okay";
+//	memory-region = <&cif_reserved>;
+};
+
+&rkcif_mipi_lvds2 {
+	status = "okay";
+
+	port {
+		cif_mipi2_in0: endpoint {
+			remote-endpoint = <&mipi2_csi2_output>;
+		};
+	};
+};
+
+&rkcif_mipi_lvds2_sditf {
+	status = "disabled";
+
+	port {
+		mipi_lvds2_sditf: endpoint {
+			remote-endpoint = <&isp0_vir0>;
+		};
+	};
+};
+
+&rkcif_mmu {
+	status = "okay";
+};
+
+&rkisp0 {
+	status = "disabled";
+};
+
+&isp0_mmu {
+	status = "disabled";
+};
+
+&rkisp0_vir0 {
+	status = "disabled";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		isp0_vir0: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&mipi_lvds2_sditf>;
+		};
+	};
+};
Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc-dual-mipi101-M101014-BE45-A1.dts
===================================================================
--- /dev/null
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc-dual-mipi101-M101014-BE45-A1.dts
@@ -0,0 +1,382 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+ *
+ */
+
+/dts-v1/;
+
+#include "roc-rk3588s-pc.dtsi"
+#include "rk3588-roc-pc-cam-8ms1m.dtsi"
+#include "rk3588-linux.dtsi"
+
+/ {
+	model = "Firefly ROC-RK3588S-PC HDMI(fydeOS)";
+	compatible = "rockchip,roc-rk3588s-pc", "rockchip,rk3588";
+
+    backlight_1: backlight-1 {
+        status = "okay";
+        compatible = "pwm-backlight";
+        enable-gpios = <&gpio4 RK_PA3 GPIO_ACTIVE_HIGH>;
+        pwms = <&pwm13 0 50000 1>;
+        brightness-levels = <
+             60  60  60  61  61  61  62  62
+             62  63  63  63  64  64  64  65
+             65  65  66  66  66  67  67  67
+             68  68  68  69  69  69  70  70
+             70  71  71  71  72  72  72  73
+             73  73  74  74  74  75  75  75
+             76  76  76  77  77  77  78  78
+             78  79  79  79  80  80  80  81
+             81  81  82  82  82  83  83  83
+             84  84  84  85  85  85  86  86
+             86  87  87  87  88  88  88  89
+             89  89  90  91  92  93  94  95
+             96  97  98  99 100 101 102 103
+            104 105 106 107 108 109 110 111
+            112 113 114 115 116 117 118 119
+            120 121 122 123 124 125 126 127
+            128 129 130 131 132 133 134 135
+            136 137 138 139 140 141 142 143
+            144 145 146 147 148 149 150 151
+            152 153 154 155 156 157 158 159
+            160 161 162 163 164 165 166 167
+            168 169 170 171 172 173 174 175
+            176 177 178 179 180 181 182 183
+            184 185 186 187 188 189 190 191
+            192 193 194 195 196 197 198 199
+            200 201 202 203 204 205 206 207
+            208 209 210 211 212 213 214 215
+            216 217 218 219 220 221 222 223
+            224 225 226 227 228 229 230 231
+            232 233 234 235 236 237 238 239
+            240 241 242 243 244 245 246 247
+            248 249 250 251 252 253 254 255
+        >;
+        default-brightness-level = <150>;
+    };
+
+    backlight: backlight {
+        status = "okay";
+        compatible = "pwm-backlight";
+        enable-gpios = <&gpio3 RK_PD4 GPIO_ACTIVE_HIGH>;
+        pwms = <&pwm14 0 50000 1>;
+        brightness-levels = <
+             60  60  60  61  61  61  62  62
+             62  63  63  63  64  64  64  65
+             65  65  66  66  66  67  67  67
+             68  68  68  69  69  69  70  70
+             70  71  71  71  72  72  72  73
+             73  73  74  74  74  75  75  75
+             76  76  76  77  77  77  78  78
+             78  79  79  79  80  80  80  81
+             81  81  82  82  82  83  83  83
+             84  84  84  85  85  85  86  86
+             86  87  87  87  88  88  88  89
+             89  89  90  91  92  93  94  95
+             96  97  98  99 100 101 102 103
+            104 105 106 107 108 109 110 111
+            112 113 114 115 116 117 118 119
+            120 121 122 123 124 125 126 127
+            128 129 130 131 132 133 134 135
+            136 137 138 139 140 141 142 143
+            144 145 146 147 148 149 150 151
+            152 153 154 155 156 157 158 159
+            160 161 162 163 164 165 166 167
+            168 169 170 171 172 173 174 175
+            176 177 178 179 180 181 182 183
+            184 185 186 187 188 189 190 191
+            192 193 194 195 196 197 198 199
+            200 201 202 203 204 205 206 207
+            208 209 210 211 212 213 214 215
+            216 217 218 219 220 221 222 223
+            224 225 226 227 228 229 230 231
+            232 233 234 235 236 237 238 239
+            240 241 242 243 244 245 246 247
+            248 249 250 251 252 253 254 255
+        >;
+        default-brightness-level = <150>;
+    };
+
+};
+
+&pwm13 {
+    pinctrl-0 = <&pwm13m1_pins>;
+    status = "okay";
+};
+
+/*
+ * mipi_dcphy1 needs to be enabled
+ * when dsi1 is enabled
+ */
+&mipi_dcphy1 {
+    status = "okay";
+};
+
+&dsi1_in_vp2 {
+    status = "disabled";
+};
+
+&dsi1_in_vp3 {
+    status = "okay";
+};
+
+&route_dsi1 {
+    status = "okay";
+    connect = <&vp3_out_dsi1>;
+};
+
+&route_hdmi0{
+    status = "disabled";
+};
+
+&dsi1 {
+    status = "okay";
+    //rockchip,lane-rate = <1000>;
+    dsi1_panel: panel@0 {
+        status = "okay";
+        compatible = "simple-panel-dsi";
+        reg = <0>;
+        backlight = <&backlight_1>;
+
+        enable-gpios = <&gpio4 RK_PA6 GPIO_ACTIVE_HIGH>;
+        reset-gpios = <&gpio4 RK_PA0 GPIO_ACTIVE_LOW>;
+
+        enable-delay-ms = <50>;
+        prepare-delay-ms = <200>;
+        reset-delay-ms = <50>;
+        init-delay-ms = <55>;
+        unprepare-delay-ms = <50>;
+        disable-delay-ms = <20>;
+        mipi-data-delay-ms = <200>;
+        size,width = <120>;
+        size,height = <170>;
+        dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_EOT_PACKET)>;
+        dsi,format = <MIPI_DSI_FMT_RGB888>;
+        dsi,lanes  = <4>;
+
+        panel-init-sequence = [
+            //39 00 04 B9 83 10 2E
+            // 15 00 02 CF FF
+            05 78 01 11
+            05 32 01 29
+            //15 00 02 35 00
+        ];
+
+        panel-exit-sequence = [
+            05 00 01 28
+            05 00 01 10
+        ];
+
+        dis1_timings0: display-timings {
+            native-mode = <&dsi1_timing0>;
+            dsi1_timing0: timing0 {
+                clock-frequency = <72600000>;//<80000000>;
+                hactive = <800>;//<768>;
+                vactive = <1280>;
+                hsync-len = <14>;   //20, 50,10
+                hback-porch = <26>; //50, 56,10
+                hfront-porch = <32>;//50, 30,180
+                vsync-len = <8>;//4
+                vback-porch = <20>;//4
+                vfront-porch = <80>;//8
+                hsync-active = <0>;
+                vsync-active = <0>;
+                de-active = <0>;
+                pixelclk-active = <0>;
+            };
+        };
+
+        ports {
+            #address-cells = <1>;
+            #size-cells = <0>;
+
+            port@0 {
+                reg = <0>;
+                panel_in_dsi1: endpoint {
+                    remote-endpoint = <&dsi1_out_panel>;
+                };
+            };
+        };
+    };
+
+    ports {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        port@1 {
+            reg = <1>;
+            dsi1_out_panel: endpoint {
+            remote-endpoint = <&panel_in_dsi1>;
+            };
+        };
+    };
+};
+
+&i2c6{
+    status = "okay";
+
+    pinctrl-names = "default";
+    pinctrl-0 = <&i2c6m3_xfer>;
+
+    hxchipset@48{
+        status = "okay";
+        compatible = "himax,hxcommon_1";
+        reg = <0x48>;
+
+        himax,rst-gpio =  <&gpio4 RK_PB3 GPIO_ACTIVE_HIGH>;
+        himax,irq-gpio = <&gpio4 RK_PA4 IRQ_TYPE_LEVEL_HIGH>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&touch_int1>;
+
+        himax,panel-coords = <0 800 0 1280>;      //触摸范围
+        himax,display-coords = <0 800 0 1280>;    //分辨率
+        report_type = <1>;
+    };
+};
+
+&pwm14 {
+    pinctrl-0 = <&pwm14m2_pins>;
+    status = "okay";
+};
+
+/*
+ * mipi_dcphy0 needs to be enabled
+ * when dsi0 is enabled
+ */
+&mipi_dcphy0 {
+    status = "okay";
+};
+
+&dsi0_in_vp2 {
+    status = "okay";
+};
+
+&dsi0_in_vp3 {
+    status = "disabled";
+};
+
+&route_dsi0 {
+    status = "okay";
+    connect = <&vp2_out_dsi0>;
+};
+
+&route_hdmi0{
+    status = "disabled";
+};
+
+&dsi0 {
+    status = "okay";
+    //rockchip,lane-rate = <1000>;
+    dsi_panel: panel@0 {
+        status = "okay";
+        compatible = "simple-panel-dsi";
+        reg = <0>;
+        backlight = <&backlight>;
+
+        enable-gpios = <&gpio1 RK_PB3 GPIO_ACTIVE_HIGH>;
+        reset-gpios = <&gpio1 RK_PA1 GPIO_ACTIVE_LOW>;
+
+        enable-delay-ms = <50>;
+        prepare-delay-ms = <200>;
+        reset-delay-ms = <50>;
+        init-delay-ms = <55>;
+        unprepare-delay-ms = <50>;
+        disable-delay-ms = <20>;
+        mipi-data-delay-ms = <200>;
+        size,width = <120>;
+        size,height = <170>;
+        dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_EOT_PACKET)>;
+        dsi,format = <MIPI_DSI_FMT_RGB888>;
+        dsi,lanes  = <4>;
+
+        panel-init-sequence = [
+            //39 00 04 B9 83 10 2E
+            // 15 00 02 CF FF
+            05 78 01 11
+            05 32 01 29
+            //15 00 02 35 00
+        ];
+
+        panel-exit-sequence = [
+            05 00 01 28
+            05 00 01 10
+        ];
+
+        disp_timings0: display-timings {
+            native-mode = <&dsi_timing0>;
+            dsi_timing0: timing0 {
+                clock-frequency = <72600000>;//<80000000>;
+                hactive = <800>;//<768>;
+                vactive = <1280>;
+                hsync-len = <14>;   //20, 50,10
+                hback-porch = <26>; //50, 56,10
+                hfront-porch = <32>;//50, 30,180
+                vsync-len = <8>;//4
+                vback-porch = <20>;//4
+                vfront-porch = <80>;//8
+                hsync-active = <0>;
+                vsync-active = <0>;
+                de-active = <0>;
+                pixelclk-active = <0>;
+            };
+        };
+
+        ports {
+            #address-cells = <1>;
+            #size-cells = <0>;
+
+            port@0 {
+                reg = <0>;
+                panel_in_dsi: endpoint {
+                    remote-endpoint = <&dsi_out_panel>;
+                };
+            };
+        };
+    };
+
+    ports {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        port@1 {
+            reg = <1>;
+            dsi_out_panel: endpoint {
+            remote-endpoint = <&panel_in_dsi>;
+            };
+        };
+    };
+};
+
+&i2c4{
+    status = "disabled";
+    pinctrl-names = "default";
+    pinctrl-0 = <&i2c4m3_xfer>;
+
+    hxchipset@48{
+        status = "okay";
+        compatible = "himax,hxcommon";
+        reg = <0x48>;
+
+        himax,rst-gpio =  <&gpio1 RK_PB2 GPIO_ACTIVE_HIGH>;
+        himax,irq-gpio = <&gpio1 RK_PA7 IRQ_TYPE_LEVEL_HIGH>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&touch_int0>;
+
+        himax,panel-coords = <0 800 0 1280>;      //触摸范围
+        himax,display-coords = <0 800 0 1280>;    //分辨率
+        report_type = <1>;
+    };
+};
+
+&pinctrl {
+    touch {
+
+        touch_int1: touch-int1 {
+            rockchip,pins = <4 RK_PA4 RK_FUNC_GPIO &pcfg_pull_up>;
+        };
+        touch_int0: touch-int0 {
+            rockchip,pins = <1 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>;
+        };
+    };
+};
Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc-ext.dtsi
===================================================================
--- /dev/null
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc-ext.dtsi
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+ *
+ */
+
+/ {
+	vcc_4g_pwr: vcc-4g-pwr {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc_4g_pwr";
+		regulator-boot-on;
+		regulator-always-on;
+		enable-active-high;
+		gpio = <&gpio3 RK_PD3 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+};
+
+&firefly_leds {
+		ext_yellow_led: ext_led1 {
+			gpios = <&gpio3 RK_PB7 GPIO_ACTIVE_HIGH>;//yellow led
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user2>;
+		};
+
+		ext_green_led: ext_led2 {
+			gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_HIGH>;//green led
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user3>;
+		};
+};
+
+&spi1 {
+	status = "okay";
+	max-freq = <48000000>;
+	dev-port = <0>;
+	pinctrl-0 = <&spi1m2_pins>;
+	num-cs = <1>;
+
+	spi_wk2xxx: spi_wk2xxx@00{
+		status = "okay";
+		compatible = "firefly,spi-wk2xxx";
+		reg = <0x00>;
+		spi-max-frequency = <10000000>;
+		//power-gpio = <&gpio2 4 GPIO_ACTIVE_HIGH>;
+		reset-gpio = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
+		irq-gpio = <&gpio3 RK_PC6 IRQ_TYPE_EDGE_FALLING>;
+		cs-gpio = <&gpio1 RK_PD3 GPIO_ACTIVE_HIGH>;
+
+		//pinctrl-0 = <&reset_gpios>,<&irq_gpios>,<&cs_gpios>;
+		//pinctrl-names = "default";
+	};
+};
+
+&can2 {
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&can2m0_pins>;
+};
+
+&pinctrl {
+	ext_leds {
+		led_user2: led-user {
+			rockchip,pins = <3 RK_PD2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+		led_user3: led-power {
+			rockchip,pins = <3 RK_PC1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
\ No newline at end of file
Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc-mipi101-M101014-BE45-A1.dts
===================================================================
--- /dev/null
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc-mipi101-M101014-BE45-A1.dts
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+ *
+ */
+
+/dts-v1/;
+
+#include "roc-rk3588s-pc.dtsi"
+#include "rk3588-roc-pc-cam-8ms1m.dtsi"
+#include "rk3588-linux.dtsi"
+#define WHICHDSI 0 /* dsi0 = 0,dsi1 = 1*/
+
+/ {
+	model = "Firefly ROC-RK3588S-PC HDMI(fydeOS)";
+	compatible = "rockchip,roc-rk3588s-pc", "rockchip,rk3588";
+
+#if WHICHDSI
+    backlight_1: backlight {
+        status = "okay";
+        compatible = "pwm-backlight";
+        enable-gpios = <&gpio4 RK_PA3 GPIO_ACTIVE_HIGH>;
+        pwms = <&pwm13 0 50000 1>;
+        brightness-levels = <
+             60  60  60  61  61  61  62  62
+             62  63  63  63  64  64  64  65
+             65  65  66  66  66  67  67  67
+             68  68  68  69  69  69  70  70
+             70  71  71  71  72  72  72  73
+             73  73  74  74  74  75  75  75
+             76  76  76  77  77  77  78  78
+             78  79  79  79  80  80  80  81
+             81  81  82  82  82  83  83  83
+             84  84  84  85  85  85  86  86
+             86  87  87  87  88  88  88  89
+             89  89  90  91  92  93  94  95
+             96  97  98  99 100 101 102 103
+            104 105 106 107 108 109 110 111
+            112 113 114 115 116 117 118 119
+            120 121 122 123 124 125 126 127
+            128 129 130 131 132 133 134 135
+            136 137 138 139 140 141 142 143
+            144 145 146 147 148 149 150 151
+            152 153 154 155 156 157 158 159
+            160 161 162 163 164 165 166 167
+            168 169 170 171 172 173 174 175
+            176 177 178 179 180 181 182 183
+            184 185 186 187 188 189 190 191
+            192 193 194 195 196 197 198 199
+            200 201 202 203 204 205 206 207
+            208 209 210 211 212 213 214 215
+            216 217 218 219 220 221 222 223
+            224 225 226 227 228 229 230 231
+            232 233 234 235 236 237 238 239
+            240 241 242 243 244 245 246 247
+            248 249 250 251 252 253 254 255
+        >;
+        default-brightness-level = <150>;
+    };
+
+#else
+
+    backlight: backlight {
+        status = "okay";
+        compatible = "pwm-backlight";
+        enable-gpios = <&gpio3 RK_PD4 GPIO_ACTIVE_HIGH>;
+        pwms = <&pwm14 0 50000 1>;
+        brightness-levels = <
+             60  60  60  61  61  61  62  62
+             62  63  63  63  64  64  64  65
+             65  65  66  66  66  67  67  67
+             68  68  68  69  69  69  70  70
+             70  71  71  71  72  72  72  73
+             73  73  74  74  74  75  75  75
+             76  76  76  77  77  77  78  78
+             78  79  79  79  80  80  80  81
+             81  81  82  82  82  83  83  83
+             84  84  84  85  85  85  86  86
+             86  87  87  87  88  88  88  89
+             89  89  90  91  92  93  94  95
+             96  97  98  99 100 101 102 103
+            104 105 106 107 108 109 110 111
+            112 113 114 115 116 117 118 119
+            120 121 122 123 124 125 126 127
+            128 129 130 131 132 133 134 135
+            136 137 138 139 140 141 142 143
+            144 145 146 147 148 149 150 151
+            152 153 154 155 156 157 158 159
+            160 161 162 163 164 165 166 167
+            168 169 170 171 172 173 174 175
+            176 177 178 179 180 181 182 183
+            184 185 186 187 188 189 190 191
+            192 193 194 195 196 197 198 199
+            200 201 202 203 204 205 206 207
+            208 209 210 211 212 213 214 215
+            216 217 218 219 220 221 222 223
+            224 225 226 227 228 229 230 231
+            232 233 234 235 236 237 238 239
+            240 241 242 243 244 245 246 247
+            248 249 250 251 252 253 254 255
+        >;
+        default-brightness-level = <150>;
+    };
+
+#endif
+
+};
+
+#if WHICHDSI
+
+&pwm13 {
+    pinctrl-0 = <&pwm13m1_pins>;
+    status = "okay";
+};
+
+/*
+ * mipi_dcphy1 needs to be enabled
+ * when dsi1 is enabled
+ */
+&mipi_dcphy1 {
+    status = "okay";
+};
+
+&dsi1_in_vp2 {
+    status = "disabled";
+};
+
+&dsi1_in_vp3 {
+    status = "okay";
+};
+
+&route_dsi1 {
+    status = "okay";
+    connect = <&vp3_out_dsi1>;
+};
+
+&route_hdmi0{
+    status = "disabled";
+};
+
+&dsi1 {
+    status = "okay";
+    //rockchip,lane-rate = <1000>;
+    dsi1_panel: panel@0 {
+        status = "okay";
+        compatible = "simple-panel-dsi";
+        reg = <0>;
+        backlight = <&backlight_1>;
+
+        enable-gpios = <&gpio4 RK_PA6 GPIO_ACTIVE_HIGH>;
+        reset-gpios = <&gpio4 RK_PA0 GPIO_ACTIVE_LOW>;
+
+        enable-delay-ms = <50>;
+        prepare-delay-ms = <200>;
+        reset-delay-ms = <50>;
+        init-delay-ms = <55>;
+        unprepare-delay-ms = <50>;
+        disable-delay-ms = <20>;
+        mipi-data-delay-ms = <200>;
+        size,width = <120>;
+        size,height = <170>;
+        dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_EOT_PACKET)>;
+        dsi,format = <MIPI_DSI_FMT_RGB888>;
+        dsi,lanes  = <4>;
+
+        panel-init-sequence = [
+            //39 00 04 B9 83 10 2E
+            // 15 00 02 CF FF
+            05 78 01 11
+            05 32 01 29
+            //15 00 02 35 00
+        ];
+
+        panel-exit-sequence = [
+            05 00 01 28
+            05 00 01 10
+        ];
+
+        disp_timings0: display-timings {
+            native-mode = <&dsi1_timing0>;
+            dsi1_timing0: timing0 {
+                clock-frequency = <72600000>;//<80000000>;
+                hactive = <800>;//<768>;
+                vactive = <1280>;
+                hsync-len = <14>;   //20, 50,10
+                hback-porch = <26>; //50, 56,10
+                hfront-porch = <32>;//50, 30,180
+                vsync-len = <8>;//4
+                vback-porch = <20>;//4
+                vfront-porch = <80>;//8
+                hsync-active = <0>;
+                vsync-active = <0>;
+                de-active = <0>;
+                pixelclk-active = <0>;
+            };
+        };
+
+        ports {
+            #address-cells = <1>;
+            #size-cells = <0>;
+
+            port@0 {
+                reg = <0>;
+                panel_in_dsi1: endpoint {
+                    remote-endpoint = <&dsi1_out_panel>;
+                };
+            };
+        };
+    };
+
+    ports {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        port@1 {
+            reg = <1>;
+            dsi1_out_panel: endpoint {
+            remote-endpoint = <&panel_in_dsi1>;
+            };
+        };
+    };
+};
+
+&i2c6{
+    status = "okay";
+
+    pinctrl-names = "default";
+    pinctrl-0 = <&i2c6m3_xfer>;
+
+    hxchipset@48{
+        status = "okay";
+        compatible = "himax,hxcommon";
+        reg = <0x48>;
+
+        himax,rst-gpio =  <&gpio4 RK_PB3 GPIO_ACTIVE_HIGH>;
+        himax,irq-gpio = <&gpio4 RK_PA4 IRQ_TYPE_LEVEL_HIGH>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&touch_int1>;
+
+        himax,panel-coords = <0 800 0 1280>;      //触摸范围
+        himax,display-coords = <0 800 0 1280>;    //分辨率
+        report_type = <1>;
+    };
+};
+
+#else
+
+&pwm14 {
+    pinctrl-0 = <&pwm14m2_pins>;
+    status = "okay";
+};
+
+/*
+ * mipi_dcphy0 needs to be enabled
+ * when dsi0 is enabled
+ */
+&mipi_dcphy0 {
+    status = "okay";
+};
+
+&dsi0_in_vp2 {
+    status = "disabled";
+};
+
+&dsi0_in_vp3 {
+    status = "okay";
+};
+
+&route_dsi0 {
+    status = "okay";
+    connect = <&vp3_out_dsi0>;
+};
+
+&route_hdmi0{
+    status = "disabled";
+};
+
+&dsi0 {
+    status = "okay";
+    //rockchip,lane-rate = <1000>;
+    dsi_panel: panel@0 {
+        status = "okay";
+        compatible = "simple-panel-dsi";
+        reg = <0>;
+        backlight = <&backlight>;
+
+        enable-gpios = <&gpio1 RK_PB3 GPIO_ACTIVE_HIGH>;
+        reset-gpios = <&gpio1 RK_PA1 GPIO_ACTIVE_LOW>;
+
+        enable-delay-ms = <50>;
+        prepare-delay-ms = <200>;
+        reset-delay-ms = <50>;
+        init-delay-ms = <55>;
+        unprepare-delay-ms = <50>;
+        disable-delay-ms = <20>;
+        mipi-data-delay-ms = <200>;
+        size,width = <120>;
+        size,height = <170>;
+        dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_EOT_PACKET)>;
+        dsi,format = <MIPI_DSI_FMT_RGB888>;
+        dsi,lanes  = <4>;
+
+        panel-init-sequence = [
+            //39 00 04 B9 83 10 2E
+            // 15 00 02 CF FF
+            05 78 01 11
+            05 32 01 29
+            //15 00 02 35 00
+        ];
+
+        panel-exit-sequence = [
+            05 00 01 28
+            05 00 01 10
+        ];
+
+        disp_timings0: display-timings {
+            native-mode = <&dsi_timing0>;
+            dsi_timing0: timing0 {
+                clock-frequency = <72600000>;//<80000000>;
+                hactive = <800>;//<768>;
+                vactive = <1280>;
+                hsync-len = <14>;   //20, 50,10
+                hback-porch = <26>; //50, 56,10
+                hfront-porch = <32>;//50, 30,180
+                vsync-len = <8>;//4
+                vback-porch = <20>;//4
+                vfront-porch = <80>;//8
+                hsync-active = <0>;
+                vsync-active = <0>;
+                de-active = <0>;
+                pixelclk-active = <0>;
+            };
+        };
+
+        ports {
+            #address-cells = <1>;
+            #size-cells = <0>;
+
+            port@0 {
+                reg = <0>;
+                panel_in_dsi: endpoint {
+                    remote-endpoint = <&dsi_out_panel>;
+                };
+            };
+        };
+    };
+
+    ports {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        port@1 {
+            reg = <1>;
+            dsi_out_panel: endpoint {
+            remote-endpoint = <&panel_in_dsi>;
+            };
+        };
+    };
+};
+
+&i2c4{
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&i2c4m3_xfer>;
+
+    hxchipset@48{
+        status = "okay";
+        compatible = "himax,hxcommon";
+        reg = <0x48>;
+
+        himax,rst-gpio =  <&gpio1 RK_PB2 GPIO_ACTIVE_HIGH>;
+        himax,irq-gpio = <&gpio1 RK_PA7 IRQ_TYPE_LEVEL_HIGH>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&touch_int0>;
+
+        himax,panel-coords = <0 800 0 1280>;      //触摸范围
+        himax,display-coords = <0 800 0 1280>;    //分辨率
+        report_type = <1>;
+    };
+};
+
+#endif
+
+&pinctrl {
+    touch {
+
+        touch_int1: touch-int1 {
+            rockchip,pins = <4 RK_PA4 RK_FUNC_GPIO &pcfg_pull_up>;
+        };
+        touch_int0: touch-int0 {
+            rockchip,pins = <1 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>;
+        };
+    };
+};
Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc.dts
===================================================================
--- /dev/null
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc.dts
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+ *
+ */
+
+/dts-v1/;
+
+#include "roc-rk3588s-pc.dtsi"
+//#include "roc-rk3588s-pc-ext.dtsi"
+
+#include "rk3588-roc-pc-cam-8ms1m.dtsi"
+
+
+/ {
+	model = "Firefly ROC-RK3588S-PC HDMI(fydeOS)";
+	compatible = "rockchip,roc-rk3588s-pc", "rockchip,rk3588";
+};
Index: kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc.dtsi
===================================================================
--- /dev/null
+++ kernel-rockchip/arch/arm64/boot/dts/rockchip/roc-rk3588s-pc.dtsi
@@ -0,0 +1,405 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
+ *
+ */
+
+#include "rk3588-firefly-port.dtsi"
+
+#define M2_SATA_OR_PCIE 1 /*1 = SATA , 0 = PCIe */
+
+/ {
+	firefly_leds: leds {
+		power_led: power {
+			gpios = <&gpio1 RK_PD5 GPIO_ACTIVE_HIGH>;//blue led
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_power>;
+		};
+
+		user_led: user {
+			gpios = <&gpio3 RK_PB2 GPIO_ACTIVE_HIGH>;//red led
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user>;
+		};
+
+		user1_led: user1 {
+			label = ":user1";
+			default-state = "off";
+			gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_HIGH>;//green led
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user1>;
+		};
+	};
+
+	fan: pwm-fan {
+		compatible = "pwm-fan";
+		#cooling-cells = <2>;
+		fan-supply = <&vcc12v_dcin>;
+		pwms = <&pwm11 0 50000 1>;
+	};
+
+	vcc3v3_pcie20: vcc3v3-pcie20 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3_pcie20";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		enable-active-high;
+		gpios = <&gpio1 RK_PD7 GPIO_ACTIVE_HIGH>;
+		startup-delay-us = <5000>;
+		vin-supply = <&vcc12v_dcin>;
+	};
+
+	vcc_3v3_sd_s0: vcc-3v3-sd-s0 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc_3v3_sd_s0";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpios = <&gpio1 RK_PA0 GPIO_ACTIVE_LOW>;
+		enable-active-low;
+		vin-supply = <&vcc_3v3_s3>;
+			regulator-state-mem {
+			regulator-off-in-suspend;
+		};
+	};
+
+	firefly_wake {
+		compatible = "firefly-wake";
+		status = "disabled";
+	};
+};
+
+/* es8323 audio codec */
+&es8388_sound {
+	status = "okay";
+	hp-det-gpio = <&gpio1 RK_PA6 GPIO_ACTIVE_LOW>;
+	hp-con-gpio = <&gpio1 RK_PA4 GPIO_ACTIVE_HIGH>;
+
+	/delete-property/ spk-con-gpio;
+
+	io-channels = <&saradc 5>;
+	io-channel-names = "hw-ver";
+
+	rockchip,audio-routing =
+			"Headphone", "LOUT2",
+			"Headphone", "ROUT2",
+			"Speaker", "LOUT1",
+			"Speaker", "ROUT1",
+			"Headphone", "Headphone Power",
+			"Headphone", "Headphone Power",
+			"LINPUT1", "Main Mic",
+			"LINPUT2", "Main Mic",
+			"RINPUT1", "Headset Mic",
+			"RINPUT2", "Headset Mic";
+	linein-type = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&hp_det>;
+};
+
+&es8388{
+	status = "okay";
+};
+
+&i2s0_8ch{
+	status = "okay";
+};
+
+/* fan */
+&pwm11 {
+	pinctrl-0 = <&pwm11m3_pins>;
+	status = "okay";
+};
+
+/* display port0 */
+&dp0_sound{
+		status = "okay";
+};
+
+&dp0 {
+	status = "okay";
+};
+
+&dp0_in_vp2 {
+	status = "okay";
+};
+
+&spdif_tx2{
+	status = "okay";
+};
+
+/* hdmi0 */
+&hdmi0 {
+	enable-gpios = <&gpio4 RK_PB2 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
+&hdmi0_in_vp0 {
+	status = "okay";
+};
+
+&hdmi0_sound {
+		status = "okay";
+};
+
+&hdptxphy_hdmi0 {
+	status = "okay";
+};
+
+&route_hdmi0{
+	status = "okay";
+};
+
+/* default use sata3.0 , pcie2.0 optional*/
+&combphy0_ps {
+	status = "okay";
+};
+
+#if M2_SATA_OR_PCIE
+&sata0 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&sata_reset>;
+	status = "okay";
+};
+#else
+&pcie2x1l2 {
+	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie20>;
+	status = "okay";
+};
+#endif
+
+/* usb3.0 */
+&combphy2_psu {
+	status = "okay";
+};
+
+&usbhost3_0 {
+	status = "okay";
+};
+
+&usbhost_dwc3_0 {
+	status = "okay";
+	dr_mode = "host";
+};
+
+&vcc5v0_host {
+	status = "okay";
+	/delete-property/ regulator-min-microvolt;
+	/delete-property/ regulator-max-microvolt;
+ 	reset-delay-us = <200000>;
+	startup-delay-us = <1200000>;
+ 	regulator-boot-on;
+	gpio = <&gpio1 RK_PB6 GPIO_ACTIVE_HIGH>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&vcc5v0_host_en>;
+};
+
+/* usb2.0 hub */
+&vcc_hub_reset {
+	status = "okay";
+	gpio = <&gpio1 RK_PB0 GPIO_ACTIVE_HIGH>;
+};
+
+/* typec0 */
+&vbus5v0_typec_pwr_en {
+		status = "okay";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpio1 RK_PB1 GPIO_ACTIVE_HIGH>;
+		vin-supply = <&vcc5v0_usb>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&typec5v_pwren>;
+};
+
+&gmac1{
+ 	snps,reset-gpio = <&gpio0 RK_PD3 GPIO_ACTIVE_LOW>;
+	tx_delay = <0x43>;
+	status = "okay";
+};
+
+/* uart7 */
+&uart7{
+	pinctrl-0 = <&uart7m2_xfer>;
+	status = "okay";
+};
+
+/* spi1 */
+&spi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi1m2_cs0 &spi1m2_pins>;
+	num-cs = <1>;
+	status = "okay";
+};
+
+/* PMIC spi2 */
+&spi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi2m2_cs0 &spi2m2_pins>;
+	num-cs = <1>;
+};
+
+&sdmmc {
+	status = "okay";
+	vmmc-supply = <&vcc_3v3_sd_s0>;
+};
+
+&i2c2 {
+	status = "okay";
+	clock-frequency = <400000>; // For others Display Port Screen
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c2m0_xfer>;
+
+	usbc0: fusb302@22 {
+		compatible = "fcs,fusb302";
+		reg = <0x22>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PC4 IRQ_TYPE_LEVEL_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&usbc0_int>;
+		vbus-supply = <&vbus5v0_typec_pwr_en>;
+		status = "okay";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				usbc0_role_sw: endpoint@0 {
+					remote-endpoint = <&dwc3_0_role_switch>;
+				};
+			};
+		};
+
+		usb_con: connector {
+			compatible = "usb-c-connector";
+			label = "USB-C";
+			data-role = "dual";
+			power-role = "dual";
+			try-power-role = "sink";
+			op-sink-microwatt = <1000000>;
+			sink-pdos =
+				<PDO_FIXED(5000, 1000, PDO_FIXED_USB_COMM)>;
+			source-pdos =
+				<PDO_FIXED(5000, 3000, PDO_FIXED_USB_COMM)>;
+
+			altmodes {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				altmode@0 {
+					reg = <0>;
+					svid = <0xff01>;
+					vdo = <0xffffffff>;
+				};
+			};
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@0 {
+					reg = <0>;
+					usbc0_orien_sw: endpoint {
+						remote-endpoint = <&usbdp_phy0_orientation_switch>;
+					};
+				};
+
+				port@1 {
+					reg = <1>;
+					dp_altmode_mux: endpoint {
+						remote-endpoint = <&usbdp_phy0_dp_altmode_mux>;
+					};
+				};
+			};
+		};
+	};
+
+	hym8563: hym8563@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		#clock-cells = <0>;
+		clock-frequency = <32768>;
+		clock-output-names = "hym8563";
+		pinctrl-names = "default";
+		pinctrl-0 = <&hym8563_int>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PB0 IRQ_TYPE_LEVEL_LOW>;
+		wakeup-source;
+		status = "okay";
+	};
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&i2c4 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c4m3_xfer>;
+};
+
+&i2c5 {
+	status = "disabled";
+};
+
+&i2c6 {
+	status = "okay";
+   	pinctrl-names = "default";
+    pinctrl-0 = <&i2c6m3_xfer>;
+};
+
+&usbdp_phy0 {
+	sbu1-dc-gpios = <&gpio4 RK_PB5 GPIO_ACTIVE_HIGH>;
+	sbu2-dc-gpios = <&gpio4 RK_PA7 GPIO_ACTIVE_HIGH>;
+};
+
+&pinctrl {
+	leds {
+		led_user: led-user {
+			rockchip,pins = <3 RK_PB2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+		led_power: led-power {
+			rockchip,pins = <1 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+		led_user1: led-user1 {
+			rockchip,pins = <3 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	headphone {
+		hp_det: hp-det {
+			rockchip,pins = <1 RK_PA6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	hym8563 {
+		hym8563_int: hym8563-int {
+			rockchip,pins = <0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
+	usb {
+		vcc5v0_host_en: vcc5v0-host-en {
+			rockchip,pins = <1 RK_PB6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	usb-typec {
+		usbc0_int: usbc0-int {
+			rockchip,pins = <0 RK_PC4 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+
+		typec5v_pwren: typec5v-pwren {
+			rockchip,pins = <1 RK_PB1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+#if M2_SATA_OR_PCIE
+	sata {
+		sata_reset:sata-reset{
+			rockchip,pins = <3 RK_PD1 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+#endif
+};
