diff --git a/Makefile b/Makefile
index d4c1ce5d725e..4ad5ddb4cc03 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ PREFIX := /usr
 SBINDIR := $(PREFIX)/sbin
 SYSCONFDIR := /etc
 RUNSTATEDIR := /var/run
-SYSTEMDSYSTEMUNITDIR := $(shell ${PKG_CONFIG} --variable=systemdsystemunitdir systemd)
+#SYSTEMDSYSTEMUNITDIR := $(shell ${PKG_CONFIG} --variable=systemdsystemunitdir systemd)
 FAN_CONTROL_SYSTEMD = systemd/fan-control.service
 VER := "1.0.0"
 
